### PR TITLE
default momentum with start=300

### PR DESCRIPTION
**Pull request by marcocuturi**
refer to #151

### CODE REVIEW

**Comment by marcocuturi** [comment]
fast test breaks on an instance with very small `epsilon` relative to distance matrix used. In that setting, 100 iterations is not enough for solution to be in a reasonable "bassin" and momentum rule diverges.... Maybe an interesting modification to consider would be to only enable adaptive momentum when iterate already has low error.

**Comment by marcocuturi** [comment]
propose to fix this by introducing an `error_threshold` parameter that checks whether error has converged sufficiently to ensure momentum helps. Set that value conservatively at `1e-1` for now.

**Comment by marcocuturi** [comment]
Observed that `momentum` can be too agressive even for "hard" instances. Fall back on an even more conservative setup were adaptive momentum is only considered after 300 iterations, with a lower `error_threshold`.

File: ott/core/momentum.py [hunk 1]
```diff
@@ -28,17 +28,25 @@ class Momentum:
...
   def weight(self, state: "sinkhorn.SinkhornState", iteration: int) -> float:
     """Compute momentum term if needed, using previously seen errors."""
-    return jnp.where(
-        iteration >= jnp.where(self.start == 0, jnp.inf, self.start),
-        self.at(state), self.value
+    if self.start == 0:
+      return self.value
+    idx = self.start // self.inner_iterations
+    weight = jnp.where(
+        iteration >= self.start,
```
Comment (michalk8) [inline ott/core/momentum.py:41]: I'd prefer this using `jax.lax.cond` + `jnp.logical_and` (no need for 2 nested where's).

**Review by michalk8 — CHANGES_REQUESTED** [review]

**Pull request merged**